cmd/doctor/doctor.go

```diff
@@ -167,7 +167,7 @@ func doctorHook(ctx context.Context, clients *shared.ClientFactory) (DoctorHookJ
 	var hookExecOpts = hooks.HookExecOpts{
 		Hook: clients.SDKConfig.Hooks.Doctor,
 	}
-	getDoctorHookJSON, err := clients.HookExecutor.Execute(hookExecOpts)
+	getDoctorHookJSON, err := clients.HookExecutor.Execute(ctx, hookExecOpts)
```

Comment:
note: This is the final result of the work - the new signature is `.Execute(ctx, opts)`.